Valeriia/Movies_database_project/src/pages/HomePage/HomePage.tsx

```diff
@@ -1,90 +1,57 @@
-import movies from "../../movies.json";
 import Header from "../../components/Header";
+import { IRootState } from "../../index";
+import { onRequestMovies } from "../../store/actions/moviesAction";
+import { IMoviesState } from "../../store/reducers/MoviesReducer";
 
-interface IHomePageState {
-  movies: IMovie[] | [];
-  isLoading: boolean;
-  tempListOfMovies: IMovie[] | [];
-  movie?: IMovie;
-  sortBy: string;
-}
-
-class HomePage extends Component<RouteComponentProps, IHomePageState> {
-  state = {
-    movies: [],
-    isLoading: false,
-    tempListOfMovies: [],
-    sortBy: "date",
-  };
+type MoviesProps = MoviesConnectProps & RouteComponentProps;
 
+class HomePage extends Component<MoviesProps, IMoviesState> {
   onSearchHandler = (searchTerm: string, filterBy: string): void => {
-    this.setState({ movie: undefined, isLoading: true });
-
-    const tempListOfMovies = this.state.movies.filter(
-      (movieItem) => movieItem[filterBy] && movieItem[filterBy] === searchTerm
-    );
-    const query = stringify({ filterBy, searchTerm });
-
-    this.setState({ tempListOfMovies, isLoading: false });
+    const queryUrl = parse(this.props.location.search) as {
+      searchTerm: string;
+      filterBy: string;
+      sortBy: string;
+    };
+    const { sortBy } = queryUrl;
+    const query = stringify({ ...queryUrl, searchTerm, filterBy });
+    const sortByType = sortBy ? sortBy : "release_date";
+    this.props.onRequestMovies(sortByType, filterBy, searchTerm);
     this.props.history.push({
       pathname: "/",
       search: query,
     });
   };
 
-  getParamFromUrlAndUpdateState = () => {
-    const query = parse(this.props.location.search) as {
+  componentDidMount = () => {
+    const queryUrl = parse(this.props.location.search) as {
       filterBy: string;
       searchTerm: string;
       sortBy: string;
     };
-    const { filterBy, searchTerm, sortBy } = query;
-    const tempListOfMovies = movies.filter((movieItem: IMovie) => {
-      if (filterBy && filterBy === "title") {
-        return movieItem.title === searchTerm;
-      } else {
-        return movieItem.genre === searchTerm;
-      }
-    });
-
-    const sortByType = sortBy ? sortBy : "date";
-
-    this.setState({
-      movies: movies,
-      isLoading: false,
-      sortBy: sortByType,
-      tempListOfMovies: tempListOfMovies.length
-        ? tempListOfMovies
-        : movies.sort((a: IMovie, b: IMovie) => {
-            if (sortByType === "date") {
-              return b.date - a.date;
-            } else {
-              return b.vote_average - a.vote_average;
-            }
-          }),
-    });
-  };
-
-  componentDidMount = () => {
-    this.setState({ isLoading: true });
-
-    setTimeout(() => {
-      this.getParamFromUrlAndUpdateState();
-    }, 1000);
+    const { searchTerm, filterBy, sortBy } = queryUrl;
+    const sortByType = sortBy ? sortBy : "release_date";
+    this.props.onRequestMovies(sortByType, filterBy, searchTerm);
```

Comment:
You have same code on 19 - 27 lines. You should move it to separate method, and call it in both places.